spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -104,12 +132,12 @@ public File getFile() throws IOException {
 
 	@Override
 	public long contentLength() throws IOException {
-		return resolve().getSize();
+		return throwExceptionForNullBlob(resolve()).getSize();
```

Comment:
Let `resolve()` throw the exception.

Reply:
This would be a different exception reason, though. This one would get thrown because resolve() might return null, on which we then call.getSize() resulting in an NPE. resolve() does this.storage.get(), which says it can return null if blob isn't found.

Reply:
I think we have to follow contract here and always throws `FileNotFoundException` if there is no resource and it isn't about `exists()` method